### PR TITLE
Enum parsing errors 2019 03 25

### PR DESCRIPTION
**Pull request by ohinds**

### CODE REVIEW

File: recidiviz/ingest/scrape/regions/us_nc_cleveland/us_nc_cleveland_scraper.py [hunk 1]
```diff
@@ -42,7 +43,10 @@ def populate_data(self, content, task: Task,
         for charge in ingest_info.get_all_charges():
             if charge.bond and charge.bond.bond_type and \
                charge.bond.bond_type == 'OUT ON BOND':
+                charge.status = ChargeStatus.PRETRIAL.value
                 charge.release_reason = ReleaseReason.BOND.value
+                charge.bond.bond_status = BondStatus.POSTED
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/regions/us_nc_cleveland/us_nc_cleveland_scraper.py]: `.value`

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py [hunk 2]
```diff
@@ -236,12 +236,54 @@ def populate_data(
...
+
+        # Handle the 'time or pay' bonds, which are cash bonds.
+        anywhere_date_strings = {
+            'DAY',
+            'WEEK',
+            'MONTH',
+            'YEAR',
+        }
+        for date_string in anywhere_date_strings:
+            overrides_builder.add(lambda bond_type: date_string in bond_type,
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: python is weird here!!! `date_string` doesn't get copied to the lambda, it's a reference, so all four of these lambdas test `'YEAR' in bond_type`. We can copy the variable to local scope by making a default arg with the same value: `lambda bond_type, date_string=date_string: date_string in bond_type`. same for endswith_date_strings

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py [hunk 3]
```diff
@@ -236,12 +236,54 @@ def populate_data(
...
+            overrides_builder.add(
+                lambda bond_type: bond_type.endswith(date_string),
+                BondType.CASH)
+
+        overrides_builder.add('ALT SENT', ChargeStatus.SENTENCED,
+                              BondType)
         overrides_builder.add('CLOSED', ChargeStatus.DROPPED)
         overrides_builder.add('DISMISSED', ChargeStatus.DROPPED, BondType)
         overrides_builder.add('O', ChargeClass.PROBATION_VIOLATION)
+        overrides_builder.add('PAROLE', ChargeStatus.SENTENCED,
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: these don't work/do anything (across bond and charge boundary)

File: recidiviz/ingest/scrape/base_scraper.py [hunk 4]
```diff
@@ -358,6 +359,10 @@ def get_enum_overrides(self) -> EnumOverrides:
                               ChargeClass.FELONY, ChargeStatus)
         overrides_builder.add(lambda status: 'MURDER' in status,
                               ChargeClass.FELONY, ChargeStatus)
+        overrides_builder.add(lambda status: 'SURETY BOND' in status,
```
Comment (jamwalla) [inline recidiviz/ingest/scrape/base_scraper.py]: we can't map across bonds and charges yet :/

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py (hunk 2 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: oh yeah, duh. i always forget about that dumb thing.

File: recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py (hunk 3 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/vendors/jailtracker/jailtracker_scraper.py]: okeee. fixed?

File: recidiviz/ingest/scrape/base_scraper.py (hunk 4 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/base_scraper.py]: dang

File: recidiviz/ingest/scrape/regions/us_nc_cleveland/us_nc_cleveland_scraper.py (hunk 1 above)
Comment (ohinds) [reply recidiviz/ingest/scrape/regions/us_nc_cleveland/us_nc_cleveland_scraper.py]: done.

**Pull request merged**